signing/src/main/java/tech/pegasys/web3signer/signing/config/DefaultArtifactSignerProvider.java

```diff
@@ -62,20 +71,41 @@ public Future<Void> load() {
     return executorService.submit(
         () -> {
           LOG.debug("Signer keys pre-loaded in memory {}", signers.size());
-
-          artifactSignerCollectionSupplier.get().stream()
-              .collect(
-                  Collectors.toMap(
-                      ArtifactSigner::getIdentifier,
-                      Function.identity(),
-                      (signer1, signer2) -> {
-                        LOG.warn(
-                            "Duplicate keys were found while loading. {}", Function.identity());
-                        return signer1;
-                      }))
-              .forEach(signers::putIfAbsent);
-
-          // for each loaded signer, load commit boost proxy signers (if any)
+          // step 1: Create copy of current signers
+          final Map<String, ArtifactSigner> oldSigners = new HashMap<>(signers);
+          // step 2: Clear current signers and then load them via ArtifactSignerCollectionSupplier
+          if (!reloadKeepStaleKeys) {
+            signers.clear();
```

Comment:
Doing this clear means there will be no signers loaded and will possibly fail some signing requests for keys aren't going to be unloaded. Can this be done without clearing the existing signers and just applying the delta to add and remove entries from the existing signer's map?